benchmark/benchmark.py

```diff
@@ -0,0 +1,659 @@
+import argparse
+import csv
+import dataclasses
+import datetime
+import enum
+import math
+import multiprocessing as pmp
+import pathlib
+import sys
+import typing
+
+import matplotlib.pyplot as plt
+import torch
+
+from gdeep.trainer.trainer import ParallelismType
+from gdeep.utility_examples.fsdp import ShardingStrategyEx
+
+sys.path.append("../examples")
+from examples import parallel_bert, parallel_orbit_5k
+
+
+class Parallelism(enum.Enum):
+    none = enum.auto()
+    fsdp_full_shard = enum.auto()
+    fsdp_shard_grad_op = enum.auto()
+    fsdp_no_shard = enum.auto()
+    pipeline = enum.auto()
```

Comment:
Parallelism is a mixture of a ParallelismType and a sharding strategy - instead one should use a composite of both.